src/test/scala/beam/agentsim/infrastructure/ChargingNetworkManagerSpec.scala

```diff
@@ -483,4 +487,23 @@ class ChargingNetworkManagerSpec
     parkingManager.ref ! Finish
     personAgent.ref ! Finish
   }
+
+  var retries = 5
+
+  override def withFixture(test: NoArgTest): Outcome = {
+    if (isRetryable(test))
+      withFixture(test, retries)
+    else
+      super.withFixture(test)
+  }
+
+  def withFixture(test: NoArgTest, count: Int): Outcome = {
+    val outcome = super.withFixture(test)
+    println(outcome.toString)
+    outcome match {
+      case Failed(_) | Canceled(_) => if (count == 1) super.withFixture(test) else withFixture(test, count - 1)
+      case other => other
+    }
+  }
```

Comment:
the same question about code duplication

Reply:
i am working on it. Please check previous comment